### ISSUE TITLE
Smudge error after updating from git 2.35.1.2 to git 2.35.2

### ISSUE THREAD

**Issue by SeriousMonk** [git-lfs/git-lfs#5110]
I had set up a git repository with git-lfs on a pc in my private network in 2019. 
I have been using it since with no problems doing git pulls and clones with the following syntax:
```
git clone X:/hostname/path
```
(where X is the letter with which I mapped the network drive on my pc)

Everything worked perfectly up until git 2.35.1.2. 
After updating to git 2.35.2 and performing git pulls and clones with either of the following syntaxes
```
git clone X:/hostname/path
git clone 192.160.2.75/path
git clone file:////hostname/path
git clone file:///hostname/path
git clone file:////192.168.2.75/path
git clone file:///192.168.2.75/path
```
I keep getting the following error:
```
Updating files: 100% (12471/12471), done.
Downloading src/DSE_menu/AHD logo ssm.jpg (13 KB)
Error downloading object: src/DSE_menu/AHD logo ssm.jpg (fe88169): Smudge error: Error downloading src/DSE_menu/AHD logo ssm.jpg (fe8816975f8a2b074bb4ef730c9d890cf5a6170a8f137b1380177a34b572aee1): batch request: missing protocol: "C:/work/repos/DSEITALIA_Master.git/info/lfs"

Errors logged to 'C:\work\repos\repo_2\.git\lfs\logs\20220912T121131.3130373.log'.
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: src/DSE_menu/AHD logo ssm.jpg: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

The logs report the following (these are with git-lfs 3.2.0, but the same can be obtained with its version used in git 2.35.2):
```
git-lfs/3.2.0 (GitHub; windows amd64; go 1.18.2)
git version 2.37.3.windows.1

$ git-lfs.exe filter-process
Error downloading object: src/DSE_menu/AHD logo ssm.jpg (fe88169): Smudge error: Error downloading src/DSE_menu/AHD logo ssm.jpg (fe8816975f8a2b074bb4ef730c9d890cf5a6170a8f137b1380177a34b572aee1): EOF

EOF
Error downloading src/DSE_menu/AHD logo ssm.jpg (fe8816975f8a2b074bb4ef730c9d890cf5a6170a8f137b1380177a34b572aee1)
github.com/git-lfs/git-lfs/v3/errors.newWrappedError
	github.com/git-lfs/git-lfs/v3/errors/types.go:229
github.com/git-lfs/git-lfs/v3/errors.Wrapf
	github.com/git-lfs/git-lfs/v3/errors/errors.go:85
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).downloadFile
	github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:120
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).Smudge
	github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:79
github.com/git-lfs/git-lfs/v3/commands.smudge
	github.com/git-lfs/git-lfs/v3/commands/command_smudge.go:128
github.com/git-lfs/git-lfs/v3/commands.filterCommand
	github.com/git-lfs/git-lfs/v3/commands/command_filter_process.go:123
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v0.0.3/command.go:766
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v0.0.3/command.go:800
github.com/git-lfs/git-lfs/v3/commands.Run
	github.com/git-lfs/git-lfs/v3/commands/run.go:107
main.main
	github.com/git-lfs/git-lfs/v3/git-lfs.go:34
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_amd64.s:1571
Smudge error
github.com/git-lfs/git-lfs/v3/errors.newWrappedError
	github.com/git-lfs/git-lfs/v3/errors/types.go:229
github.com/git-lfs/git-lfs/v3/errors.NewSmudgeError
	github.com/git-lfs/git-lfs/v3/errors/types.go:315
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).Smudge
	github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:88
github.com/git-lfs/git-lfs/v3/commands.smudge
	github.com/git-lfs/git-lfs/v3/commands/command_smudge.go:128
github.com/git-lfs/git-lfs/v3/commands.filterCommand
	github.com/git-lfs/git-lfs/v3/commands/command_filter_process.go:123
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v0.0.3/command.go:766
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v0.0.3/command.go:800
github.com/git-lfs/git-lfs/v3/commands.Run
	github.com/git-lfs/git-lfs/v3/commands/run.go:107
main.main
	github.com/git-lfs/git-lfs/v3/git-lfs.go:34
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_amd64.s:1571
```

This only happens if the master repository is on a different pc within the local network. If the master repository is on the same pc as the one performing the clone everything works perfectly.

So I downgraded back to git 2.35.1.2 and everything works perfectly again.

It seems that up until git 2.35.1.2 it works and it stops working in the followin versions (including the latest one 2.37.3).

### Steps to reproduce

1. Install latest git version 2.37.3
2. Create a master repository on a device on the local network
3. Upload some files to be tracked by git lfs
4. Try cloning the repository from a different device on the same network

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  Would you mind including the output of `git lfs env` from the working and non-working environments?

I wonder if you're encountering some consequences from the key change in Git between v2.35.1 and v2.35.2, which was to address CVE-2022-24765:

https://github.com/git/git/compare/v2.35.1...v2.32.2
https://github.com/git-for-windows/git/compare/v2.35.1.windows.2...v2.35.2.windows.1

**Comment by SeriousMonk** [comment]
Hi, would you need the ```git lfs env``` with the latest version of git installed or with the one I'm using now (2.35.1.2)?

**Comment by SeriousMonk** [comment]
Hi,

Here is the env when I had the latest git version (that caused the smudge error):
```
Environment:
LocalWorkingDir=D:\Users\FRANZ\Desktop\repo_1
LocalGitDir=D:\Users\FRANZ\Desktop\repo_1\.git
LocalGitStorageDir=D:\Users\FRANZ\Desktop\repo_1\.git
LocalMediaDir=D:\Users\FRANZ\Desktop\repo_1\.git\lfs\objects
LocalReferenceDirs=
TempDir=D:\Users\FRANZ\Desktop\repo_1\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=D:\Users\FRANZ\Desktop\repo_1\.git\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_DIR=D:/Users/FRANZ/Desktop/repo_1/.git
GIT_EXEC_PATH=D:/Programs/Git/mingw64/libexec/git-core
```

and here si the environment now that I'm back on git version 2.35.1.2:
```
git-lfs/3.0.2 (GitHub; windows amd64; go 1.17.2)
git version 2.35.1.windows.2

Endpoint=file:///x:/DSEITALIA_Master (auth=none)
LocalWorkingDir=C:\work\repos\repo_1
LocalGitDir=C:\work\repos\repo_1\.git
LocalGitStorageDir=C:\work\repos\repo_1\.git
LocalMediaDir=C:\work\repos\repo_1\.git\lfs\objects
LocalReferenceDirs=
TempDir=C:\work\repos\repo_1\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=C:\work\repos\repo_1\.git\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=D:/Program Files/Git_old/mingw64/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

```

Finally here is the environment on the host with the master repository. 
```
git-lfs/3.2.0 (GitHub; windows amd64; go 1.18.2)
git version 2.37.1.windows.1

LocalWorkingDir=
LocalGitDir=
LocalGitStorageDir=
LocalMediaDir=lfs\objects
LocalReferenceDirs=
TempDir=lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Comment by chrisd8088** [comment]
Thanks!  It's interesting, I think, that not only is the `Endpoint` line missing (which comes from `git config remote.origin.url` or a similar command) but also the `git config filter.lfs.*` lines at the end of the output when you've upgraded to a recent Git.

I should have asked before, but what do you get as output when you run `git config -l` with the new Git vs. the old one?  (Take care to remove any authorization strings before pasting it here.)

Also, is the Git repository directory owned by a different user than the one you're running the commands with?

I have a suspicion that you might need to set a global Git `safe.directory` configuration.  This was added as part of the security fix in Git in v2.35.2 that I mentioned before.  Here's the [description](https://git-scm.com/docs/git-config/2.35.2#Documentation/git-config.txt-safedirectory), which notes that Git won't parse a config file in another user's directory now:
> These config entries specify Git-tracked directories that are considered safe even if they are owned by someone other than the current user. By default, Git will refuse to even parse a Git config of a repository owned by someone else, let alone run its hooks, and this config setting allows users to specify exceptions, e.g. for intentionally shared repositories...

So you might check for whether your Git working tree directory is owned by another user, and whether `git config -l` produces any output with the new Git.

**Comment by SeriousMonk** [comment]
Bingo!

After running
```
git config --global --add safe.directory "%(prefix)///BUSSOLINO/Siti WEB/DSEITALIA_Master"
```
The command
```
git clone x:/DSEITALIA_Master repo_4
```
(which was giving the smudge error) is now completing successfully
```
Cloning into 'repo_4'...
done.
Updating files: 100% (12471/12471), done.
Filtering content: 100% (11239/11239), 4.17 GiB | 28.26 MiB/s, done.
```

Thank you very much for your help.
